src/app/modules/spcp/spcp.util.ts

```diff
@@ -299,10 +295,11 @@ export const getRedirectTarget = (
   formId: string,
   authType: FormAuthType.SP | FormAuthType.CP,
   isPersistentLogin?: boolean,
+  encodedQuery?: string,
 ): string =>
   `/${formId},${
     // Need to cast to boolean because undefined is allowed as a valid value
     // We are not following corppass's official spec for
     // the target parameter
     authType === FormAuthType.SP ? !!isPersistentLogin : false
-  }`
+  },${encodedQuery ?? ''}`
```

Comment:
nit: should we not add encoded query if it does not exist at all instead of an empty string?